assertj-core/src/main/java/org/assertj/core/error/ShouldHaveCause.java

```diff
@@ -45,9 +46,11 @@ private ShouldHaveCause(Throwable actualCause, Throwable expectedCause) {
           "but type was:%n" +
           "  %s%n" +
           "and message was:%n" +
-          "  %s.",
+          "  %s.%n" +
+          "Throwable that failed the check:%n" +
+          "%s",
           expectedCause.getClass().getName(), expectedCause.getMessage(),
-          actualCause.getClass().getName(), actualCause.getMessage());
+          actualCause.getClass().getName(), actualCause.getMessage(), getStackTrace(actualCause));
```

Comment:
that's not correct, we want to have the actual's stack trace, not actual's cause.

the way to address that is to pass actual in `shouldHaveCause` instead of  `actualCause`, this will require change to `Throwables#assertHasCause` that passes `actualCause`.